SwaprHQ/stackly-ui
# [STK-90][REFACTOR] - Modal component dividers, DOM tree improvements, docs enhancement, Nextjs errors fixing

## Fixes: [STK-90](https://linear.app/swaprdev/issue/STK-90/modal-component-dividers)

### Description

* Dividers are now conditional styles for Modal sub-components. Handling them with `divide-y` from Tailwind is a no-go for now as we don't have dividers consistency across current modals.
* Created a new React Context to handle all the dapp Modals.
* Optimized dapp current Contexts by stabilizing their referenced values


  #### Minor stuff:
* Remove unneeded/unused files like duplicated icons and the whole `packages/app/app/ui` folder.
* Add `Troubleshooting` and `Code Guidelines` to the README, with some examples.
* General improvements in the DOM tree, removing unneeded elements, updating imports, and module renaming.
* Removed a few Nextjs warnings regarding hydration issues.

### Preview
https://refactor-stk-90-modal-compon.stackly-ui.pages.dev/

### How to test the changes

1) Pull this branch
2) Run the project locally
3) Go to Stackly landing page, both app behavior and app look&feel should be the same, maybe a bit faster
